core/src/main/scala/kafka/utils/ByteBoundedBlockingQueue.scala

```diff
@@ -17,17 +17,20 @@
 
 package kafka.utils
 
-import java.util.concurrent.atomic.AtomicInteger
-import java.util.concurrent.{TimeUnit, LinkedBlockingQueue}
+import java.util.concurrent.atomic.AtomicLong
+import java.util.concurrent.{LinkedBlockingQueue, TimeUnit}
 
 /**
  * A blocking queue that have size limits on both number of elements and number of bytes.
  */
-class ByteBoundedBlockingQueue[E] (val queueNumMessageCapacity: Int, val queueByteCapacity: Int, sizeFunction: Option[(E) => Int])
+class ByteBoundedBlockingQueue[E] (val queueNumMessageCapacity: Int, val queueByteCapacity: Long, sizeFunction: Option[(E) => Long])
     extends Iterable[E] {
+  if (sizeFunction == null || sizeFunction.isEmpty) {
+    throw new IllegalArgumentException("size function must be provided")
+  }
```

Comment:
If we want to make it mandatory, then we should just change `sizeFunction` to be `E => Long`. `Option` only makes sense if we want it to be optional.